### ISSUE TITLE
Tag

### ISSUE THREAD

**Issue by Ivanecho** [files-community/Files#15530]
### Description

The sorting doesn't make sense when I adopt the "sorting with tags". I had added tags to folders and the icon shown up. But when I sorted the folders with tags, there are only one sort, which is 'unclassified' in the interface.
![微信图片_20240602153600](https://github.com/files-community/Files/assets/171474679/4ca846f7-a0ce-4ea5-9862-5d6b1707f663)

### Steps To Reproduce

1. Added tags to folders
2. Switching the sorting way to 'tags'
3. 'nothing happened'

### Requirements

Fix crash that occurs when sorting with tags

### Files Version

3.4.1.0

### Windows Version

10.0.22631.3593

### Log File

[debug.log](https://github.com/user-attachments/files/15524503/debug.log)

**Comment by Josh65-2201** [comment]
Thanks for the report, can you see if this is an issue in the preview version. https://files.community/download

You can export your settings in Files settings > Advanced > Export > Import them in preview version

**Comment by Ivanecho** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Thank you for the suggestion. But it still doesn't work in the preview version (see the screenshot).
        Btw I love this software especially the functions of column view and tags. Please help me with this problem so that I can continue to use it.
    
        
            
        
    
    
        
        

        
    

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        6/2/2024 19:26
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [files-community/Files] Tag (Issue #15530)
        
      
    
  
  
Thanks for the report, can you see if this is an issue in the preview version. https://files.community/download
You can export your settings in Files settings > Advanced > Export > Import them in preview version

**Comment by Ivanecho** [comment]
it still doesn't work in the preview version

**Comment by Josh65-2201** [comment]
Thanks for checking

**Comment by yaira2** [comment]
Is this only an issue in the Columns View?

**Comment by Ivanecho** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        It occurs in all sorts of views. Its arrangement was correct immediately after I switched to the column view, but it went wrong again when I exited and reopened it.
        
    
        
            
        
    
    
        
        

        
    

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        6/19/2024 09:59
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [files-community/Files] Bug: Tags aren't grouped correctly (Issue #15530)
        
      
    
  
  
Is this only an issue in the Columns View?

**Comment by TuTAH1** [comment]
I have the same bug, if understand the issue correctly – I set Tags for grouping and it displays correctly:
 
![image](https://github.com/user-attachments/assets/1a85081f-e968-4299-a3f3-205ea48184ec)

Then if I do any action refreshing interface, like adding a new tag, removing one, change any of folder's layout settings or simply close and open Files – it's now displaying all fildes/folders in "Untagged" group.

![image](https://github.com/user-attachments/assets/f4927b1b-4afb-4356-84dc-53e818abac25)

Now, the only way to fix it, is to go to settings, layout and change the default grouping method (even if it means to switch it to something and back into the same method as it was) – now everything grouped by tags, but again, ANY interface refresh and it's all "Untagged" again.

**Comment by saphtea** [comment]
This is frustrating !! There's a lot of awesome QoL improvements in Files but this is the one reason I switched to it and it's bugged 😭. I appreciate y'all working on this software to begin with tho, hope this gets fixed soon :)

**Comment by ferrariofilippo** [comment]
Isn't this the same as #11759?

**Comment by Josh65-2201** [comment]
I don't think so this one is just about them not being sorted correctly the other one is about them not displaying the file/folder
